qtkeychain/keychain_unix.cpp

```diff
@@ -186,6 +186,20 @@ static KeyringBackend detectKeyringBackend()
 
 static KeyringBackend getKeyringBackend()
 {
+    // Check if user wants to override detection logic
+    QByteArray backendOverride = qgetenv("QTKEYCHAIN_BACKEND");
+    if (backendOverride == "libsecret") {
+        return Backend_LibSecretKeyring;
+    } else if (backendOverride == "gnome") {
+        return Backend_GnomeKeyring;
+    } else if (backendOverride == "kwallet4") {
+        return Backend_Kwallet4;
+    } else if (backendOverride == "kwallet5") {
+        return Backend_Kwallet5;
+    } else if (backendOverride == "kwallet6") {
+        return Backend_Kwallet6;
+    }
```

Comment:
I think this should go at the start of `detectKeyringBackend()`, for two reasons:
1. So that all of the backend selection logic would be in one place.
2. So that the environment override value would be saved in the `static KeyringBackend backend` variable, same as without the override.

Reply:
Okay, I moved it.